scripts/DeployAgTokenSideChainMultiBridge.s.sol

```diff
@@ -0,0 +1,158 @@
+            } else {
+                proxyAdmin = _chainToContract(chainId, ContractType.ProxyAdmin);
+            }
+            if (vm.keyExistsJson(json, ".coreBorrow")) {
+                coreBorrow = vm.parseJsonAddress(json, ".coreBorrow");
+            } else {
+                coreBorrow = _chainToContract(chainId, ContractType.CoreBorrow);
+            }
+        } else {
+            proxyAdmin = _chainToContract(chainId, ContractType.ProxyAdmin);
+            coreBorrow = _chainToContract(chainId, ContractType.CoreBorrow);
+        }
+        ILayerZeroEndpoint lzEndpoint = _lzEndPoint(chainId);
+        address expectedAddress;
+        if (vm.envExists("EXPECTED_ADDRESS")) {
+            expectedAddress = vm.envAddress("EXPECTED_ADDRESS");
+        } else {
+            // TODO compute the expected address once one of the address has been deployed
+            if (keccak256(abi.encodePacked(stableName)) == keccak256("USD")) {
+                expectedAddress = _chainToContract(CHAIN_ETHEREUM, ContractType.AgUSD);
+            }
+            if (keccak256(abi.encodePacked(stableName)) == keccak256("EUR")) {
+                expectedAddress = address(0x00001063a5d3A9d5f1B03e848fa788aae1d98C5c);
+            }
+        }
+
+        vm.startBroadcast(deployerPrivateKey);
+
+        AgTokenSideChainMultiBridge agTokenImpl = new AgTokenSideChainMultiBridge();
+        console.log("AgTokenSideChainMultiBridge Implementation deployed at", address(agTokenImpl));
+
+        ImmutableCreate2Factory create2Factory = ImmutableCreate2Factory(IMMUTABLE_CREATE2_FACTORY_ADDRESS);
+        address computedAddress = create2Factory.findCreate2Address(salt, initCode);
+        console.log("AgTokenSideChainMultiBridge Proxy Supposed to deploy: %s", computedAddress);
+
+        require(computedAddress == expectedAddress, "Computed address does not match expected address");
+
+        AgTokenSideChainMultiBridge agToken = AgTokenSideChainMultiBridge(create2Factory.safeCreate2(salt, initCode));
+        TransparentUpgradeableProxy(payable(address(agToken))).upgradeTo(address(agTokenImpl));
+        TransparentUpgradeableProxy(payable(address(agToken))).changeAdmin(proxyAdmin);
+        console.log("AgTokenSideChainMultiBridge Proxy deployed at", address(agToken));
+
+        Treasury treasuryImpl = new Treasury();
+        console.log("Treasury Implementation deployed at", address(treasuryImpl));
+
+        Treasury treasuryProxy = Treasury(
+            address(
+                _deployUpgradeable(
+                    proxyAdmin,
+                    address(treasuryImpl),
+                    abi.encodeWithSelector(Treasury.initialize.selector, coreBorrow, address(agToken))
+                )
+            )
+        );
+        console.log("Treasury Proxy deployed at", address(treasuryProxy));
+
+        agToken.initialize(string.concat(stableName, "A"), string.concat(stableName, "A"), address(treasuryProxy));
```

Comment:
So stableName is `EUR`, `USD` etc right?

Reply:
yes